optaplanner-core/src/main/java/org/optaplanner/core/impl/score/director/AbstractScoreDirector.java

```diff
@@ -501,6 +514,30 @@ public void assertWorkingScoreFromScratch(Score workingScore, Object completedAc
         }
     }
 
+    @Override
+    public void assertExpectedUndoMoveScore(Move move, Score beforeMoveScore) {
+        Score undoScore = calculateScore();
+        if (!undoScore.equals(beforeMoveScore)) {
+            logger.trace("        Corruption detected. Diagnosing...");
+            // TODO PLANNER-421 Avoid undoMove.toString() because it's stale (because the move is already done)
+            String undoMoveString = "Undo(" + move + ")";
+            // Precondition: assert that are probably no corrupted score rules.
```

Comment:
Typo? Looks like it should be something like "assert there are" or "assert that there are".